app/models/user.rb

```diff
@@ -44,6 +46,12 @@ class User < ApplicationRecord
   enum gender: { male: 0, female: 1, other: 2 }
 
   has_many :spots, dependent: :destroy
+  has_many :first_matches, class_name: 'Match', foreign_key: 'first_user_id'
```

Comment:
What do you think about adding `dependent: :destroy` option? If an user no longer exist the match will be orphan